metricbeat/module/vsphere/datastore/datastore.go

```diff
@@ -51,6 +54,25 @@ func New(base mb.BaseMetricSet) (mb.MetricSet, error) {
 	return &MetricSet{ms}, nil
 }
 
+type metricData struct {
+	perfMetrics map[string]interface{}
+	assetsName  assetNames
+}
+
+type assetNames struct {
+	outputVmNames []string
+	outputHsNames []string
+}
+
+// Define metrics to be collected
+var metricNames = []string{
```

Comment:
@niraj-elastic can we make this changes in data store as well?
https://github.com/elastic/beats/pull/40429#issuecomment-2298397761

1. Use single map for metricsNames
2. Use context to exit out of dst for loop.

Reply:
right!